File: testsupport/nexus-testsuite-support/src/main/java/org/sonatype/nexus/testsuite/support/Filter.java
/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
package org.sonatype.nexus.testsuite.support;

import java.util.Map;

/**
 * Filter placeholders. For example could take as input "some.company:foo:${project.version}" and replaces the
 * placeholder "${project.version}" with the actual value "some.company:foo:1.0" by looking up the version from test
 * pom.
 *
 * @since 2.2
 */
public interface Filter
{

  /**
   * Filters placeholders.
   *
   * @param context filtering context. Cannot be null.
   * @param value   value to be filtered. Cannot be null.
   * @return filtered value. If null the filter is not considered in filtering chain.
   */
  String filter(final Map<String, String> context, String value);

}

File: components/nexus-rapture/src/main/resources/static/rapture/NX/util/Filter.js
/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Open Source Version is distributed with Sencha Ext JS pursuant to a FLOSS Exception agreed upon
 * between Sonatype, Inc. and Sencha Inc. Sencha Ext JS is licensed under GPL v3 and cannot be redistributed as part of a
 * closed source work.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext*/

/**
 * Filter
 *
 * @since 3.3
 */
Ext.define('NX.util.Filter', {
  singleton: true,

  /**
   * Util to build a div for empty search results.
   *
   * @param searchString
   * @param emptyTemplate
   * @returns {string}
   */
  buildEmptyResult: function(searchString, emptyTemplate) {
    var encoded = Ext.util.Format.htmlEncode(searchString);
    return '<div class="x-grid-empty">' + emptyTemplate.replace(/\$filter/, encoded) + '</div>';
  }
});
